gis/gccview/gcc_layer_refresh.py

```diff
@@ -0,0 +1,664 @@
+                                                              pk = sql.Identifier(primary_key),
+                                                              cols = insert_column,
+                                                              excl_cols = excluded_column,
+                                                              date = sql.Identifier(date)))
+                    LOGGER.info('Updated table %s', output_table)
+                except Exception:
+                    # pass exception to function
+                    logging.exception("Failed to UPSERT")
+                    # rollback the previous transaction before starting another
+                    con.rollback()
+                    successful_execution = False
+            
+            # if table does not exist -> create a new one and add to audit list
+            else:
+                try:
+                    cur.execute(sql.SQL("alter table {schema}.{temp_table} rename to {tablename}; comment on table {schema}.{tablename} is 'last updated: {date}'").format(schema = sql.Identifier(schema_name), temp_table = sql.Identifier(temp_table_name), tablename = sql.Identifier(output_table), date = sql.Identifier(date)))
+
+                    cur.execute(sql.SQL("select {schema}.audit_table({schema}.{tablename})").format(schema = sql.Identifier(schema_name),
+                                                                                                     tablename = sql.Identifier(output_table)))
+                    LOGGER.info('New table %s created and added to audit table list', output_table)
+                except Exception:
+                    # pass exception to function
+                    logging.exception("Failed to create new table")
+                    # rollback the previous transaction before starting another
+                    con.rollback()
+                    successful_execution = False
+            
+            # And then drop the temp table (if exists)
+            cur.execute(sql.SQL("drop table if exists {schema}.{temp_table}").format(schema = sql.Identifier(schema_name),
+                                                                                         temp_table = sql.Identifier(temp_table_name)))
+    return successful_execution
+#-------------------------------------------------------------------------------------------------------
+'''
+@click.command()
+@click.option('--mapserver_n', '-ms', help = 'Mapserver number, e.g. cotgeospatial_2 will be 2', type = int)
+@click.option('--layer_id', '-ly', help = 'Layer id', type = int)
+@click.option('--schema_name', '-s', help = 'Name of destination schema', type = str)
+@click.option('--is_audited', '-a', help = 'Whether the table is supposed to be audited (T) or partitioned (F)', type = bool)
+'''
+#-------------------------------------------------------------------------------------------------------
```

Comment:
Not that I'm really sure we still need command-line interfaces (but they can useful!).

The here pipeline has a decent example of having the click decorated function being a wrapper around a "main" Python function. (note there's some complexity in that program due to trying to allow for running the sub-functions of the pipeline that's not relevant here. AKA anything involving the `ctx` object)

Reply:
I think these decorators are a bit outdated. In the final implementation, only `manual_get_layer`, which is called in `__main__`, should be decorated with `click`